### PR TITLE
sd-stub: some minor refactoring of the cpio/TPM code

### PR DESCRIPTION
**Pull request by poettering**
This is split out of #24458, but is a nice refactoring that makes sense on its own, too, and fixes a (minor) bug.

### CODE REVIEW

File: src/boot/efi/cpio.c [hunk 1]
```diff
@@ -304,6 +304,47 @@ static EFI_STATUS pack_cpio_trailer(
...
+                                buffer_size,
+                                tpm_description,
+                                &m);
+                if (err != EFI_SUCCESS) {
+                        log_error_stall(L"Unable to add initrd TPM measurement for PCR %u (%s), ignoring: %r", tpm_pcr[i], tpm_description, err);
+                        measured = false;
+                        continue;
+                }
+
+                measured = measured < 0 ? m : (measured && m);
```
Comment (keszybz) [inline src/boot/efi/cpio.c]: Maybe write this as
```C
if (measured != false)
     measured = m;
```
I think that's a bit easier to unwrap.

File: src/boot/efi/cpio.c [hunk 2]
```diff
@@ -486,7 +486,7 @@ EFI_STATUS pack_cpio(
         *ret_buffer = NULL;
         *ret_buffer_size = 0;
 
-        if (ret_measured)
+        if (ret_measured && n_tpm_pcr > 0)
                 *ret_measured = true;
```
Comment (keszybz) [inline src/boot/efi/cpio.c]: Hmm, doesn't this violate the rule that the output variable should be always set?

**Review by keszybz — CHANGES_REQUESTED** [review]

File: src/boot/efi/cpio.c (hunk 2 above)
Comment (poettering) [reply src/boot/efi/cpio.c]: yes, i am an idiot ;-)

**Comment by poettering** [comment]
pushed a new version, with the two points raised addressed. thanks!

**Review by keszybz — APPROVED** [review]
LGTM.

**Pull request merged**